source/class/qx/lang/String.js

```diff
@@ -146,6 +149,57 @@ qx.Bootstrap.define("qx.lang.String",
     },
 
 
+    /**
+     * Detects whether the string is all upper case
+     * @param {String} str 
+     * @returns {Boolean}
+     */
+     isUpperCase(str) {
+      if (str.length == 0)
+        return false;
+      for (let i = 0; i < str.length; i++) {
+        let type = qx.lang.String.__characterTypes[str[i]];
+        if (type !== "upper")
+          return false;
+      }
+      return true;
```

Comment:
how about this

```suggestion
      return qx.lang.String.__characterRx.upper.test(str);
```

Reply:
i assumed that regex would be massively slower, but didn't try it.  I'll work up some quick tests

Reply:
There is also the problem of international language support ... with the proposed implementation, the method would have to called `isEnglishUppercase`  :)

Reply:
fair comment.  I've compared jsfiddle https://jsfiddle.net/srcgux9n/ and in Chrome there is between 35% and 50% overhead for regex, but it still does 1,000,000 regex in under 50ms - Firefox RegEx is faster than maps!